src/pyqasm/algorithms/solovay_kitaev/optimizer.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+from pyqasm.elements import BasisSet
+
+IDENTITY_WEIGHT_GROUP = {
+    BasisSet.CLIFFORD_T: {
+        "h": {
+            "group": "h",
+            "weight": 0.5
+        },
+        "s": {
+            "group": "s-t",
+            "weight": 0.25
+        },
+        "t": {
+            "group": "s-t",
+            "weight": 0.125
+        },
+        "sdg": {
+            "group": "sdg-tdg",
+            "weight": 0.25
+        },
+        "tdg": {
+            "group": "sdg-tdg",
+            "weight": 0.125
+        },
+    }
+}
+
+def optimize_gate_sequnce(seq: list[str], target_basis_set):
+    target_identity_weight_group = IDENTITY_WEIGHT_GROUP[target_basis_set]
+    while True:
+        current_group = None
+        current_weight = 0
+        start_index = 0
+        changed = False
+
+        for i, gate_name in enumerate(seq):
+            gate = target_identity_weight_group[gate_name]
```

Comment:
add a check for the existence of `gate` in the `arget_identity_weight_group` dict